### DISCUSSION TITLE
Reword "Dependency Dashboard approval" to something simpler?

### DISCUSSION THREAD

**Discussion by HonkingGoose** [renovatebot/renovate#24948] [Suggest an Idea]
### Tell us more.

@aaronadamsCA commented: [^1]

> Improved text when `dependencyDashboardApproval` is enabled would also be helpful.
> 
> I'm encouraging a team to adopt Renovate. I've updated the onboarding branch with a simple config that would not create any pull requests (except for vulnerability alerts). But the "What to Expect" description still threatens to create PRs.
> Config
> Onboarding PR description
> 
> "Enable Renovate Dependency Dashboard approval workflow" is jargon that would benefit from more simple language. "Renovate will create 21 Pull Requests" is simple language that sounds scary because it is incorrect.

I agree that "Dependency Dashboard approval workflow" is jargon. I'd like to use a simpler term, if possible. 😄 

## Brainstorming new names

The simplest explanation would be something like: _Only get updates from Renovate after you select them from the Dependency Dashboard._ But that's too long as a config option name.

### Drop `dependency` part?

We could maybe drop the `Dependency` part of the `DependencyDashboard` name, but then users may ask which dashboard it refers to:

- The Dependency Dashboard issue, or
- Mend's Dashboard with the global overview of repositories

### Silly list of names

- `onlyGetUpdatesAfterSelectingFromDashboardFirst`
- `selectUpdatesFromDashboardBeforeGettingThem`
- `approveUpdatesFromDashboardBeforeGettingThem`
- `selectUpdatesFromDashboardFirst`
- `approveUpdatesFromDashboardFirst`

## Renaming todos

_If_ we decide to rename, here are some todos:

- [ ] Rename `dependencyDashboardApproval` config option to simpler term
- [ ] Write config migration to migrate from old term to new term
- [ ] Rewrite onboarding PR body text to use simpler language, update fixtures
- [ ] Update all docs that refer to `dependencyDashboardApproval` config option
- [ ] Update all docs that have the string `dependency dashboard approval`

[^1]: https://github.com/renovatebot/renovate/issues/18939#issuecomment-1735635773

**Comment by rarkins** [comment]
Related feature/option: we also have prCreation=approval, which controls the _PR_ creation via dashboard.

The main benefit of using `dependencyDashboardApproval` is that it then sorts next to `dependencyDashboard`.

But maybe we could have a new term of `branchCreation` and `approval` or `dashboard-approval` is one of the options.

**Reply by HonkingGoose** [reply]
> Related feature/option: we also have prCreation=approval, which controls the _PR_ creation via dashboard.

Our docs don't explain what `prCreation=approval` does, I'm fixing it with PR:

- #24949

> The main benefit of using `dependencyDashboardApproval` is that it then sorts next to `dependencyDashboard`.

Agree!

> But maybe we could have a new term of `branchCreation` and `approval` or `dashboard-approval` is one of the options.

Why do we have `prCreation=approval` _and_ `dependencyDashboardApproval=true`? I would expect us to only have `dependencyDashboardApproval=true`... I'm confused now. 😄

**Reply by rarkins** [reply]
Usually you wouldn't use both.

dependencyDashboardApproval=true means don't do anything at all unless approved in the dashboard
dependencyDashboardApproval=false + prCreation=approval means push the branch/commit, let tests run, but only create the PR on-demand. The idea was that people could see if it passed or not before asking for the PR

**Reply by HonkingGoose** [reply]
> But maybe we could have a new term of `branchCreation` and `approval` or `dashboard-approval` is one of the options.

So do you want to rename `prCreation` to `branchCreation` then? 

I like `dashboard-approval` best, because it says _where_ to approve things.

**Reply by rarkins** [reply]
No, pr creation and branch creation are separate steps. I'm suggesting that dependencyDashboardApproval=true could be renamed as part of a new branchCreation config.

But really I'm not sure this is all that confusing to people? The original discussion you linked to was not about the person confused by approval - they understood it. They just said that the onboarding PR body did not did not reflect the approval steps

**Reply by HonkingGoose** [reply]
> No, pr creation and branch creation are separate steps. I'm suggesting that dependencyDashboardApproval=true could be renamed as part of a new branchCreation config.

Oh okay! I'll let you create an issue, if you want. 😉 

> But really I'm not sure this is all that confusing to people? The original discussion you linked to was not about the person confused by approval - they understood it. They just said that the onboarding PR body did not did not reflect the approval steps

If we're happy with the term `Dependency Dashboard Approval workflow` we can keep it, and only change the onboarding PR body text. I'll let you create the issue for that also. 😉
